### PR TITLE
a8n: Rename Campaign.changesetCreationStatus to .status

### PR DESCRIPTION
**Pull request by mrnugget**
As requested here: https://github.com/sourcegraph/sourcegraph/pull/7346#discussion_r365306633

Also makes a lot of sense in the context of updating a Campaign.

I also made it non-nullable.

@eseliger @felixfbecker can you check whether my TypeScript changes are okay?

### CODE REVIEW

File: web/src/enterprise/campaigns/detail/CampaignDetails.tsx [hunk 1]
```diff
@@ -307,7 +307,7 @@ export const CampaignDetails: React.FunctionComponent<Props> = ({
     const status = campaign
         ? campaign.__typename === 'CampaignPlan'
             ? campaign.status
```
Comment (eseliger) [inline web/src/enterprise/campaigns/detail/CampaignDetails.tsx]: this ternary is redudant now, can just be `campaign.status` directly

**Comment by mrnugget** [comment]
This is also a breaking change — I don't think anybody relies on this yet, but we need to add it to the changelog, right?

**Comment by eseliger** [comment]
hmmm technically

File: web/src/enterprise/campaigns/detail/CampaignDetails.tsx (hunk 1 above)
Comment (mrnugget) [reply web/src/enterprise/campaigns/detail/CampaignDetails.tsx]: This was me search/replacing on the CLI. Feel free to edit this branch 😉

**Comment by eseliger** [comment]
@christinaforney do you think the entry in the changelog is enough documentation of this breaking change here? We don't think anyone has been using it yet.

**Comment by mrnugget** [comment]
@eseliger It looks like the web snapshot tests need to be updated.

**Comment by eseliger** [comment]
oops, done

**Comment by mrnugget** [comment]
Regarding backwards compatibility: I don't have a strong opinion on whether to merge this PR or not. I do think it makes sense to have the same name for the field on both types and I bet we don't have any users of this API yet, but on the other hand: we're making a theoretically breaking API change to get rid of a single ternary statement. Is that worth it? If @felixfbecker and @eseliger say yes and Christina signs off on the CHANGELOG entry I'll merge this :)

**Comment by felixfbecker** [comment]
When we designed this API, I remember we said to not spend too much time on the details and that everything is in alpha and can still be changed. So I want to operate under that model :)
As soon as we have consumers of the API that model needs to change of course, @christinaforney can tell us :)
Alternatively, I think maybe we should think about recording some non-webapp/bext GraphQL field usage in server pings to make these decisions easier.

**Comment by mrnugget** [comment]
> When we designed this API, I remember we said to not spend too much time on the details and that everything is in alpha and can still be changed. So I want to operate under that model :)
> As soon as we have consumers of the API that model needs to change of course, @christinaforney can tell us :)

Yep, it's absolutely fine from my side. I'd rather change it now than in a few months. Just wanted to confirm.

**Comment by christinaforney** [comment]
A changelog entry is sufficient at this point in time! Go for it :)

**Pull request merged**